src/common/debug.c

```diff
@@ -222,3 +222,30 @@ DP_UUID(const void *uuid)
 	thread_uuid_str_buf_idx = (thread_uuid_str_buf_idx + 1) % DF_UUID_MAX;
 	return buf;
 }
+
+#define DF_KEY_MAX		8
+#define DF_KEY_STR_SIZE		64
+
+static __thread int thread_key_buf_idx;
+static __thread char thread_key_buf[DF_KEY_MAX][DF_KEY_STR_SIZE];
+
+char *
+daos_key2str(daos_key_t *key)
+{
+	char *buf = thread_key_buf[thread_key_buf_idx];
+
+	if (!key->iov_buf || key->iov_len == 0) {
+		strcpy(buf, "<NULL>");
+	} else {
+		int len = min(key->iov_len, DF_KEY_STR_SIZE - 1);
+
+		if (isprint(*(char *)key->iov_buf)) {
+			strncpy(buf, key->iov_buf, len);
+			buf[len] = 0;
+		} else {
+			strcpy(buf, "????");
+		}
+	}
+	thread_key_buf_idx = (thread_key_buf_idx + 1) % DF_KEY_MAX;
```

Comment:
did not get why need DF_KEY_MAX number of thread local key buf? one per thread is not enough?

Reply:
if we call this function for more than once in one D_DEBUG, this is going to be useful